### ISSUE TITLE
Default filter and hide filtered items

### ISSUE THREAD

**Issue by danrspencer** [DestinyItemManager/DIM#10188]
### Proposed change

These are probably two changes but I've lumped them together because I think they enable some powerful functionality.

1. Default filter

Simply a saved filter that is set by default when DIM is opened.

2. Hide filtered items

Currently DIM grays out non-matching items; I'd like an option to instead completely hide non-matching items. With 100s of items in the vault simply graying out filtered items still leaves a lot of visual noise.

It's possible this should only apply the items in the vault to avoid the confusing situation of looking at your character screen and not seeing some of those items in DIM. 

There should be some text stating something like "hiding X items" so that people don't forget they've got a filter applied.

### How does this fit into your workflow?

My primary use case for this is that I would like to be able to open DIM and completely ignore every item I've tagged as junk. I don't care about them until I'm out of vault space so I don't want to see them. 

But that's my personal use case; the beauty of these two simple changes is that it lets people really customise their DIM default page. Someone else may be a primarily PvE player so they only want to see guns where they've added a PvE note.

tldr; I'd like to be able to open DIM and see only what I care about, but quickly expand to everything that we currently have.

**Comment by bhollis** [comment]
Thanks for the suggestion.

> Default filter

This is one that's been asked for before, and we won't be adding it as an option. Use tags (such as Archive) and item sort configs to display items in different places.

> Hide filtered items

This is another that's been asked and we won't be adding. If you want to see just items that match a search, open the Search Results page by hitting enter in the search field or clicking the search results icon.

**Comment by danrspencer** [comment]
Do you mind if I ask why you won’t be adding them?

Would you be open to a PR to add them or are they strictly features you don’t want to be part of DIM?

**Comment by bhollis** [comment]
We would not accept a PR to add those features. This has been a long standing part of DIM's design and we have a policy to avoid adding options where possible.
